### DISCUSSION TITLE
[Feature] Search for a location or metadata in smart search instead of specifically using the filters

### DISCUSSION THREAD

**Discussion by SerAlbi** [immich-app/immich#8682] [General]
It would be useful and google's photo-like to have the smart search bar respond to both metadata and ai context tags. 
As an example in google photo I would often search for a specific city or place and all the photos taken at that place would show up. 

In immich instead we have to search with the metadata tag m:city or select the city from a dropdown of existing cities which is not very user friendly. 
At least there should be an option to directly write the city you want to search instead of selecting it from a drop-down.

The same should apply to every metadata, as an example if i search for "mom" immich should show me photos of my mom instead of requiring me to select "mom" from the specific 'People' filter pill.

**Comment by vidyava** [comment]
To expand on this (because it felt like a duplicate to suggest it separately, since it would need a unified smart search interface), it would also be nice to have the capability to logically OR/NOT filters with the unified search, since the current search doesn't seem to support either. 

Some examples of where it would be useful:
Pictures with my parents:
`person:me AND (person:mom OR person:dad)`

Pictures of bodies of water (though CLIP could likely handle this one):
`tag:ocean OR tag:river OR tag:pond`

A picture fishing with my dad probably around 2017, but I'm not sure I tagged it or the exact date so show me both subsets:
`person:dad AND (tag:fishing OR year:2017)`

A picture of my mom traveling:
`person:mom AND NOT city:homecity`

Photos that Immich has identified as a specific person, or photos that I've tagged as a person since we currently can't batch associate unmatched photos with people. This could also be handled by just having the search work on both person names and tag names simultaneously:
`person:mom OR tag:mom`

Forgive me if it's frowned on to link other implementations that I've liked, but HomeGallery has a pretty extensive token based search interface that I would love to have in Immich: [Link](https://docs.home-gallery.org/search/index.html)

**Comment by bartekchrup** [comment]
I absolutely approve of this, when I simply type a country name into the search it uses "Context" rather than "Country" which means lets say if i search "Spain", the results are of places that look Spanish, rather than having the location metadata in Spain.

If you serach for a location on Google Photos it auto-completes locations, which if clicked opens all pictures taken in that place.

**Comment by janbro1** [comment]
I would like to add a concrete use case for general boolean and negative search filters.

My goal is to find assets that match a semantic search query but have not yet been manually categorized:

`context:meme AND NOT tag:Meme AND NOT is:trashed`

This is useful as a curation workflow:

1. Smart Search identifies likely memes.
2. Assets already carrying the `Meme` tag are excluded.
3. Assets in the trash are excluded.
4. The remaining results can be reviewed and tagged in bulk.

I do not think this should be implemented as a special case for tags. A general solution would be much more useful:

* `AND`, `OR`, and `NOT` across contextual search and metadata filters
* inclusion and exclusion of tags, people, albums, locations, ratings, and similar fields
* grouping with parentheses where needed
* the same semantics in the web UI, mobile apps, and API
* trashed assets excluded by default, with an explicit option to include them

A possible UI implementation could use three states for filter chips:

* not selected
* include
* exclude

Alternatively, a documented query syntax would also work.

The same feature would support many other maintenance workflows, for example:

* dogs without the `Dog` tag
* documents not yet assigned to an album
* photos containing a person but missing a corresponding manual tag
* travel photos outside a particular location
* assets without an expected organizational tag

Would this discussion be the appropriate place to track this broader boolean-search requirement?

Related discussion regarding broader search and filtering improvements: [#11777](https://github.com/immich-app/immich/discussions/11777)
